rotkehlchen/chain/evm/decoding/odos/common.py

```diff
@@ -0,0 +1,202 @@
+                tx_hash=context.transaction.tx_hash,
+                timestamp=context.transaction.timestamp,
+                event_type=HistoryEventType.SPEND,
+                event_subtype=HistoryEventSubType.FEE,
+                asset=held_asset,
+                balance=Balance(amount=amount),
+                location_label=sender,
+                notes=f'Spend {amount} {held_asset.symbol_or_name()} as an {self.label} fee',
+                counterparty=self.counterparties()[0].identifier,
+                address=self.router_address,
+            )
+            for held_asset, amount in router_holding_amount.items()
+            if amount != ZERO
+        ]
+
+    def decode_swap(
+            self,
+            context: 'DecoderContext',
+            sender: 'ChecksumEvmAddress',
+            input_tokens: dict[str, FVal],
+            output_tokens: dict[str, FVal],
+    ) -> 'DecodingOutput':
+        """Decodes swaps done using an Odos v1/v2 router"""
+        in_events, out_events = [], []
+        for event in context.decoded_events:
+            if (
+                ((input_amount := input_tokens.get(event.asset.identifier)) == event.balance.amount) or  # noqa: E501
+                input_amount == ZERO  # this means https://docs.odos.xyz/product/sor/v2/#max-balance-swapping  # noqa: E501
+            ) and (
+                (
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ) or (
+                    event.event_type == HistoryEventType.TRADE and
+                    event.event_subtype == HistoryEventSubType.SPEND
+                )
+            ):
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.SPEND
+                event.notes = f'Swap {event.balance.amount} {event.asset.symbol_or_name()} in {self.label}'  # noqa: E501
+                event.address = self.router_address
+                event.counterparty = self.counterparties()[0].identifier
+                out_events.append(event)
+
+            elif output_tokens.get(event.asset.identifier) == event.balance.amount and (
+                (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ) or (
+                    event.event_type == HistoryEventType.TRADE and
+                    event.event_subtype == HistoryEventSubType.RECEIVE
+                )
+            ):
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.RECEIVE
+                event.notes = f'Receive {event.balance.amount} {event.asset.symbol_or_name()} as the result of a swap in {self.label}'  # noqa: E501
+                event.address = self.router_address
+                event.counterparty = self.counterparties()[0].identifier
```

Comment:
Both for this and for the label in other places we put it as part of the constructor. Why not do the same here? Putting another constructor argument and saving it in memory duplicates info?

Reply:
yeah that was the reason